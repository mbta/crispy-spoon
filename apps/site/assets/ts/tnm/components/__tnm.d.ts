import { EnhancedRoute, Mode, StopWithDirections } from "../../__v3api";

export interface StopWithRoutes {
  stop: StopWithDirections;
  routes: RouteGroup[];
  distance: string;
}

export interface RouteGroup {
  group_name: Mode;
  routes: EnhancedRoute[];
}
