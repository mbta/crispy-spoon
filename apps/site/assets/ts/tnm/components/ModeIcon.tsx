import React, { ReactElement } from "react";
import GreenLineIconSmall from "../../../static/images/icon-green-line-small.svg";
import RedLineIconSmall from "../../../static/images/icon-red-line-small.svg";
import OrangeLineIconSmall from "../../../static/images/icon-orange-line-small.svg";
import BlueLineIconSmall from "../../../static/images/icon-blue-line-small.svg";
import MattapanSmall from "../../../static/images/icon-mattapan-line-small.svg";
import CommuterRailIconSmall from "../../../static/images/icon-mode-commuter-rail-small.svg";
import BusIconSmall from "../../../static/images/icon-mode-bus-small.svg";
import FerryIconSmall from "../../../static/images/icon-mode-ferry-small.svg";
import StationSmall from "../../../static/images/icon-circle-t-small.svg";
import SubwaySmall from "../../../static/images/icon-mode-subway-small.svg";

interface Props {
  type: string;
}

const icons = {
  "green_line-small": GreenLineIconSmall,
  "red_line-small": RedLineIconSmall,
  "orange_line-small": OrangeLineIconSmall,
  "blue_line-small": BlueLineIconSmall,
  "mattapan_line-small": MattapanSmall,
  "commuter_rail-small": CommuterRailIconSmall,
  "bus-small": BusIconSmall,
  "ferry-small": FerryIconSmall,
  "subway-small": SubwaySmall
};

const ModeIcon = ({ type }: Props): ReactElement<HTMLElement> => {
  // @ts-ignore
  const icon = icons[`${type}-small`] || StationSmall;

  return (
    <span
      className="m-tnm-sidebar__mode-icon"
      // eslint-disable-next-line react/no-danger
      dangerouslySetInnerHTML={{ __html: icon }}
    />
  );
};

export default ModeIcon;
