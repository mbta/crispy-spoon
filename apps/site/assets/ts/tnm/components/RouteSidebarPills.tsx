import React, { ReactElement } from "react";
import { Stop } from "../../__v3api";
import { clickStopPillAction, Dispatch } from "../state";
import { handleReactEnterKeyPress } from "../../helpers/keyboard-events-react";

interface Props {
  selectedStop: Stop | undefined;
  dispatch: Dispatch;
  showPill: boolean;
}

const RouteSidebarPills = (props: Props): ReactElement<HTMLElement> | null => {
  const { dispatch, selectedStop, showPill } = props;
  const onClickPill = (): void => dispatch(clickStopPillAction());
  return showPill && selectedStop ? (
    <>
      <div className="m-tnm-sidebar__separator" />
      <span
        role="button"
        tabIndex={0}
        className="m-tnm-sidebar__pill"
        onClick={onClickPill}
        onKeyPress={e => handleReactEnterKeyPress(e, onClickPill)}
        aria-label={`Remove filtering by ${selectedStop.name}`}
      >
        <span className="m-tnm-sidebar__pill-name">{selectedStop.name}</span>
        <span className="m-tnm-sidebar__pill-close fa fa-times-circle" />
      </span>
    </>
  ) : null;
};

export default RouteSidebarPills;
