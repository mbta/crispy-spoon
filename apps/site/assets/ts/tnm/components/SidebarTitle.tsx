import React, { ReactElement } from "react";
import { handleReactEnterKeyPress } from "../../helpers/keyboard-events-react";
import { clickViewChangeAction, Dispatch } from "../state";

interface Props {
  dispatch: Dispatch;
  viewType: string;
}

const onClickViewChange = (dispatch: Dispatch): void =>
  dispatch(clickViewChangeAction());

const SidebarTitle = ({
  viewType,
  dispatch
}: Props): ReactElement<HTMLElement> => {
  const onClick = (): void => onClickViewChange(dispatch);
  return (
    <div className="m-tnm-sidebar__header">
      <div className="m-tnm-sidebar__header-title">
        <h2>{`Nearby ${viewType}`}</h2>
        <div
          className="m-tnm-sidebar__view-change"
          role="button"
          tabIndex={0}
          onClick={onClick}
          onKeyPress={e => handleReactEnterKeyPress(e, onClick)}
        >
          Change view
        </div>
      </div>
    </div>
  );
};

export default SidebarTitle;
