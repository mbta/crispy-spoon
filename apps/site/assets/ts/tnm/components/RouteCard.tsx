/* eslint-disable react/prefer-stateless-function */
import React, { ReactElement } from "react";
import useSWR from "swr";
import StopCard from "../../components/StopCard";
import {
  Direction,
  StopWithDirections,
  RouteWithStopsWithDirections,
  Alert
} from "../../__v3api";
import { Dispatch } from "../state";
import { directionIsEmpty } from "../../components/Direction";
import { modeByV3ModeType } from "../../components/ModeFilter";
import RouteCardHeader from "../../components/RouteCardHeader";
import { isABusRoute } from "../../models/route";
import { isHighSeverityOrHighPriority } from "../../models/alert";
import { fetchJsonOrThrow, isFetchFailed } from "../../helpers/fetch-json";

interface Props {
  route: RouteWithStopsWithDirections;
  dispatch: Dispatch;
}

const everyDirectionIsEmpty = (directions: Direction[]): boolean =>
  directions.every(directionIsEmpty);

export const fetchAlerts = async (routeId: string): Promise<Alert[]> => {
  const results = await fetchJsonOrThrow<Alert[]>(
    `/api/alerts?route_ids=${routeId}`
  );
  if (isFetchFailed(results)) {
    throw new Error(
      `Failed to fetch Alert information: ${results.status} ${results.statusText}`
    );
  }

  return results.filter(isHighSeverityOrHighPriority);
};

const routeIsEmpty = (route: RouteWithStopsWithDirections): boolean =>
  route.stops_with_directions.every(stop =>
    everyDirectionIsEmpty(stop.directions)
  );

const filterStops = (
  route: RouteWithStopsWithDirections
): StopWithDirections[] => {
  // show the closest two stops for bus, in order to display both inbound and outbound stops

  const count = isABusRoute(route.route) ? 2 : 1;
  return route.stops_with_directions.slice(0, count);
};

const RouteCard = ({
  route,
  dispatch
}: Props): ReactElement<HTMLElement> | null => {
  const mode = modeByV3ModeType[route.route.type];

  const { data: alerts } = useSWR<Alert[]>(`${route.route.id}`, fetchAlerts);

  if (routeIsEmpty(route)) {
    return null;
  }

  return (
    <div className="m-tnm-sidebar__route" data-mode={mode}>
      <RouteCardHeader route={route.route} alerts={alerts || []} />
      {filterStops(route).map(
        stopWithDirections =>
          !everyDirectionIsEmpty(stopWithDirections.directions) && (
            <StopCard
              key={stopWithDirections.stop.id}
              stop={stopWithDirections.stop}
              distance={stopWithDirections.distance}
              directions={stopWithDirections.directions}
              route={route.route}
              dispatch={dispatch}
            />
          )
      )}
    </div>
  );
};

export default RouteCard;
