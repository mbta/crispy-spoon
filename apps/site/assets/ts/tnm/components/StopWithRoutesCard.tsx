import React, { ReactElement } from "react";
import { Mode, EnhancedRoute, Stop } from "../../__v3api";
import { RouteGroup } from "./__tnm";
import { Dispatch } from "../state";
import ModeIcon from "./ModeIcon";
import { useSMDown } from "../../helpers/media-breakpoints-react";
import { buttonProps } from "../../components/StopCard";

export const renderRoutesLabel = (
  routes: EnhancedRoute[],
  type: Mode
): ReactElement<HTMLElement> =>
  type === "commuter_rail" ? (
    <a href={routes[0].href}>Commuter Rail</a>
  ) : (
    <span>
      {type === "bus" ? "Bus: " : null}
      {routes.map((route, i: number) => (
        <React.Fragment key={route.id}>
          <a href={route.href}>{route.name}</a>
          {i !== routes.length - 1 ? ", " : ""}
        </React.Fragment>
      ))}
    </span>
  );

export const renderRoutes = (
  routes: EnhancedRoute[],
  type: Mode
): ReactElement<HTMLElement> => (
  <div key={type} className="m-tnm-sidebar__stop-card-description">
    <span className="m-tnm-sidebar__stop-route-name">
      <ModeIcon type={type} />
      {renderRoutesLabel(routes, type)}
    </span>
  </div>
);

interface Props {
  stop: Stop;
  routes: RouteGroup[];
  dispatch: Dispatch;
  distance: string;
}

const StopWithRoutesCard = ({
  stop,
  routes,
  distance,
  dispatch
}: Props): ReactElement<HTMLElement> => {
  const isSmallBreakpoint = useSMDown();
  const containerProps = !isSmallBreakpoint
    ? buttonProps(dispatch, stop.id)
    : {};

  return (
    <div className="m-tnm-sidebar__stop-card" {...containerProps}>
      <div className="m-tnm-sidebar__stop-card-header">
        <a className="m-tnm-sidebar__stop-card-name" href={stop.href}>
          {stop.name}
        </a>
        <div className="m-tnm-sidebar__stop-distance">{distance}</div>
      </div>
      {routes.map(
        ({ group_name: groupName, routes: routesForStop }: RouteGroup) =>
          renderRoutes(routesForStop, groupName)
      )}
    </div>
  );
};

export default StopWithRoutesCard;
