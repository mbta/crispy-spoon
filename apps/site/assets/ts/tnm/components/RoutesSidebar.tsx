import React, { ReactElement, useEffect } from "react";
import { modeByV3ModeType } from "../../components/ModeFilter";
import { Dispatch } from "../state";
import { Alert, Mode, RouteWithStopsWithDirections, Stop } from "../../__v3api";
import RouteCard from "./RouteCard";
import RouteSidebarPills from "./RouteSidebarPills";
import ModeFilterContainer from "./ModeFilterContainer";
import SidebarTitle from "./SidebarTitle";
import { fetchJsonOrThrow, isFetchFailed } from "../../helpers/fetch-json";
import { isHighSeverityOrHighPriority } from "../../models/alert";
import useSWR from "swr";

interface Props {
  data: RouteWithStopsWithDirections[];
  dispatch: Dispatch;
  selectedStopId: string | null;
  shouldFilterStopCards: boolean;
  selectedStop: Stop | undefined;
  selectedModes: Mode[];
  emptyMessage: ReactElement<HTMLElement>;
}

interface FilterOptions {
  stopId: string | null;
  modes: Mode[];
}

export const fetchAlerts = async (routeId: string): Promise<Alert[]> => {
  const results = await fetchJsonOrThrow<Alert[]>(
    `/api/alerts?route_ids=${routeId}`
  );

  if (isFetchFailed(results)) {
    // 404s here are a known failure mode, see finder_api.ex#get_trip_info
    throw new Error(
      `Failed to fetch Alert information: ${results.status} ${results.statusText}`
    );
  }

  let alerts: Alert[] = [];
  results.filter(isHighSeverityOrHighPriority);
  for (let i = 0; i < results.length; i++) {
    alerts.push(results[i] as Alert);
  }
  return alerts;
};

export const isUndefined = (arg: any): any => {
  if (arg == undefined) {
    return [];
  }
  return arg;
};

//fetchAlerts("Orange").then(function(val) {console.log(val[0] instanceof Alert)});
// export const fetchAlerts = async (
//   routeId: string,
// ): Promise<Alert[]> => {
//   let alerts: Alert[] = [];
//   window.fetch &&
//   window
//     .fetch(`/api/alerts?route_ids=${routeId}`)
//     .then(response => {
//       if (response.ok) return response.json();
//       throw new Error(response.statusText);
//     })
//     .then(result => {
//         for (let alert in result) {
//           alerts.push(JSON.parse(alert) as Alert);
//         }
//     console.log(alerts);

//       return alerts;
//     })
//     console.log(alerts);
//     return alerts;
// };

const filterDataByModes = (
  data: RouteWithStopsWithDirections[],
  { modes }: FilterOptions
): RouteWithStopsWithDirections[] => {
  // if there are no selections or all selections, do not filter
  if (modes.length === 0 || modes.length === 3) {
    return data;
  }
  return data.filter(route =>
    modes.reduce((accumulator: boolean, mode: Mode) => {
      if (accumulator === true) {
        return accumulator;
      }
      return modeByV3ModeType[route.route.type] === mode;
    }, false)
  );
};

const filterDataByStopId = (
  data: RouteWithStopsWithDirections[],
  { stopId }: FilterOptions
): RouteWithStopsWithDirections[] => {
  if (stopId === null) {
    return data;
  }
  return data.reduce(
    (
      accumulator: RouteWithStopsWithDirections[],
      route: RouteWithStopsWithDirections
    ): RouteWithStopsWithDirections[] => {
      // eslint-disable-next-line camelcase
      const stops = route.stops_with_directions.filter(
        // eslint-disable-next-line camelcase
        stop_with_directions => stop_with_directions.stop.id === stopId
      );
      if (stops.length === 0) {
        return accumulator;
      }
      // eslint-disable-next-line camelcase
      return accumulator.concat({ ...route, stops_with_directions: stops });
    },
    []
  );
};

export const filterData = (
  data: RouteWithStopsWithDirections[],
  selectedStopId: string | null,
  selectedModes: Mode[],
  shouldFilter: boolean
): RouteWithStopsWithDirections[] => {
  if (shouldFilter === false) {
    return data;
  }

  const options: FilterOptions = {
    stopId: selectedStopId,
    modes: selectedModes
  };

  return [filterDataByStopId, filterDataByModes].reduce(
    (accumulator, fn) => fn(accumulator, options),
    data
  );
};

const RoutesSidebar = ({
  data,
  dispatch,
  selectedModes,
  selectedStopId,
  selectedStop,
  shouldFilterStopCards,
  emptyMessage
}: Props): ReactElement<HTMLElement> => {
  const filteredData = filterData(
    data,
    selectedStopId,
    selectedModes,
    shouldFilterStopCards
  );

  // const alerts = new Map<String, Alert[]>();

  // useEffect(() => {
  // filteredData.length > 0 ? filteredData.forEach(route => alerts.set(route.route.id, (useSWR<Alert[]>(`/api/alerts?route_ids=${"Orange"}`, fetchAlerts).data))) : [];
  // console.log(alerts);
  // });

  return (
    <div className="m-tnm-sidebar">
      <div className="m-tnm-sidebar__fixed-header">
        <div className="m-tnm-sidebar__fixed-header-inner">
          <ModeFilterContainer
            selectedModes={selectedModes}
            dispatch={dispatch}
          />
          <RouteSidebarPills
            selectedStop={selectedStop}
            showPill={shouldFilterStopCards}
            dispatch={dispatch}
          />
        </div>
      </div>
      <div className="m-tnm-sidebar__inner">
        <SidebarTitle dispatch={dispatch} viewType="Routes" />

        <div className="m-tnm-sidebar__cards">
          {filteredData.length > 0
            ? filteredData.map(route => (
                <RouteCard
                  key={route.route.id}
                  route={route}
                  dispatch={dispatch}
                  alerts={isUndefined(
                    useSWR<Alert[]>(
                      `/api/alerts?route_ids=${"Orange"}`,
                      fetchAlerts
                    ).data
                  )}
                />
              ))
            : emptyMessage}
        </div>
      </div>
    </div>
  );
};
export default RoutesSidebar;
