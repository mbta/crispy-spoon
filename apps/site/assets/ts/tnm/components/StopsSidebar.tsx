import React, { ReactElement } from "react";
import { Mode } from "../../__v3api";
import { StopWithRoutes } from "./__tnm";
import { Dispatch } from "../state";
import SidebarTitle from "./SidebarTitle";
import StopWithRoutesCard from "./StopWithRoutesCard";
import ModeFilterContainer from "./ModeFilterContainer";
import stopIncludesModes from "../helpers/stop-includes-modes";

interface Props {
  data: StopWithRoutes[];
  dispatch: Dispatch;
  selectedStopId: string | null;
  shouldFilterStopCards: boolean;
  selectedModes: Mode[];
  emptyMessage: ReactElement<HTMLElement>;
}

interface FilterOptions {
  stopId: string | null;
  modes: Mode[];
}

const filterDataByStopId = (
  data: StopWithRoutes[],
  { stopId }: FilterOptions
): StopWithRoutes[] => {
  if (stopId === null) {
    return data;
  }
  const stopWithRoutes = data.find(d => d.stop.id === stopId);
  return stopWithRoutes ? [stopWithRoutes] : data;
};

const filterDataByModes = (
  data: StopWithRoutes[],
  { modes }: FilterOptions
): StopWithRoutes[] => data.filter(stop => stopIncludesModes(stop, modes));

export const filterData = (
  data: StopWithRoutes[],
  selectedStopId: string | null,
  selectedModes: Mode[],
  shouldFilter: boolean
): StopWithRoutes[] => {
  if (shouldFilter === false) {
    return data;
  }

  const options: FilterOptions = {
    stopId: selectedStopId,
    modes: selectedModes
  };

  return [filterDataByStopId, filterDataByModes].reduce(
    (accumulator, fn) => fn(accumulator, options),
    data
  );
};

const StopsSidebar = ({
  dispatch,
  data,
  selectedStopId,
  selectedModes,
  shouldFilterStopCards,
  emptyMessage
}: Props): ReactElement<HTMLElement> => {
  const filteredData = filterData(
    data,
    selectedStopId,
    selectedModes,
    shouldFilterStopCards
  );
  return (
    <div className="m-tnm-sidebar" id="tnm-sidebar-by-stops">
      <div className="m-tnm-sidebar__fixed-header">
        <div className="m-tnm-sidebar__fixed-header-inner">
          <ModeFilterContainer
            selectedModes={selectedModes}
            dispatch={dispatch}
          />
        </div>
      </div>
      <div className="m-tnm-sidebar__inner">
        <SidebarTitle dispatch={dispatch} viewType="Stops" />
        <div className="m-tnm-sidebar__cards">
          {filteredData.length > 0
            ? filteredData.map(({ stop, routes, distance }) => (
                <StopWithRoutesCard
                  key={stop.id}
                  stop={stop}
                  routes={routes}
                  dispatch={dispatch}
                  distance={distance}
                />
              ))
            : emptyMessage}
        </div>
      </div>
    </div>
  );
};

export default StopsSidebar;
