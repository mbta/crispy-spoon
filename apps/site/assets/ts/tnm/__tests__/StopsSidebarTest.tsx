import React from "react";
import renderer from "react-test-renderer";
import StopsSidebar, { filterData } from "../components/StopsSidebar";
import { createReactRoot } from "../../app/helpers/testUtils";
import { importData, importRealtimeResponse } from "./helpers/testUtils";
import { StopWithRoutes } from "../components/__tnm";
import { transformStops } from "../helpers/process-realtime-data";

const realtimeData = importRealtimeResponse();
const stopsWithDistances = importData();
const stopsWithRoutes = transformStops(
  stopsWithDistances.distances,
  [],
  realtimeData
);

describe("StopsSidebar", () => {
  it("it renders", () => {
    createReactRoot();
    const tree = renderer
      .create(
        <StopsSidebar
          data={stopsWithRoutes}
          selectedStopId={null}
          shouldFilterStopCards={false}
          dispatch={() => {}}
          selectedModes={[]}
          emptyMessage={<div>No data</div>}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("it returns null when there isn't data", () => {
    createReactRoot();
    const tree = renderer
      .create(
        <StopsSidebar
          data={[]}
          selectedStopId={null}
          shouldFilterStopCards={false}
          dispatch={() => {}}
          selectedModes={[]}
          emptyMessage={<div>No data</div>}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("filterData", () => {
  it("should filter by stop ID", () => {
    const selectedStopId = stopsWithRoutes[0].stop.id;

    expect(stopsWithRoutes).toHaveLength(1);

    const filteredData = filterData(stopsWithRoutes, selectedStopId, [], true);

    expect(filteredData).toHaveLength(1);

    // Only the selected stop should be shown
    expect(
      filteredData.every(
        ({ stop: { id } }: StopWithRoutes) => id === selectedStopId
      )
    );
  });

  it("should not filter if stop is not found", () => {
    const selectedStopId = "something";

    expect(stopsWithRoutes).toHaveLength(1);

    const filteredData = filterData(stopsWithRoutes, selectedStopId, [], true);

    // Every stop should be shown
    expect(filteredData).toHaveLength(1);
  });

  it("should filter by modes", () => {
    expect(stopsWithRoutes).toHaveLength(1);

    const filteredBusData = filterData(stopsWithRoutes, null, ["bus"], true);
    expect(filteredBusData).toHaveLength(1);

    const filteredRailData = filterData(
      stopsWithRoutes,
      null,
      ["commuter_rail"],
      true
    );
    expect(filteredRailData).toHaveLength(1);

    const filteredSubwayData = filterData(
      stopsWithRoutes,
      null,
      ["subway"],
      true
    );
    expect(filteredSubwayData).toHaveLength(1);
  });
});
