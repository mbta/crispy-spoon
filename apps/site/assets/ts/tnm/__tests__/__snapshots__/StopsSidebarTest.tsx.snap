// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StopsSidebar it renders 1`] = `
<div
  className="m-tnm-sidebar"
  id="tnm-sidebar-by-stops"
>
  <div
    className="m-tnm-sidebar__fixed-header"
  >
    <div
      className="m-tnm-sidebar__fixed-header-inner"
    >
      <div
        className="m-tnm-sidebar__filter-bar"
      >
        <span
          className="m-tnm-sidebar__filter-header u-small-caps"
        >
          Filter
        </span>
        <div
          className="c-mode-filter__filter-btn-group"
        >
          <button
            aria-label="add filter by subway"
            className="btn btn-secondary btn-sm m-tnm-sidebar__filter-btn inactive"
            id="mode-button__subway"
            onClick={[Function]}
            type="button"
          >
            <span
              className="m-tnm-sidebar__mode-icon"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "SVG",
                }
              }
            />
            Subway
          </button>
          <button
            aria-label="add filter by bus"
            className="btn btn-secondary btn-sm m-tnm-sidebar__filter-btn inactive"
            id="mode-button__bus"
            onClick={[Function]}
            type="button"
          >
            <span
              className="m-tnm-sidebar__mode-icon"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "SVG",
                }
              }
            />
            Bus
          </button>
        </div>
        <div
          className="c-mode-filter__filter-btn-group"
        >
          <button
            aria-label="add filter by commuter_rail"
            className="btn btn-secondary btn-sm m-tnm-sidebar__filter-btn inactive"
            id="mode-button__commuter_rail"
            onClick={[Function]}
            type="button"
          >
            <span
              className="m-tnm-sidebar__mode-icon"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "SVG",
                }
              }
            />
            Rail
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="m-tnm-sidebar__inner"
  >
    <div
      className="m-tnm-sidebar__header"
    >
      <div
        className="m-tnm-sidebar__header-title"
      >
        <h2>
          Nearby Stops
        </h2>
        <div
          className="m-tnm-sidebar__view-change"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex={0}
        >
          Change view
        </div>
      </div>
    </div>
    <div
      className="m-tnm-sidebar__cards"
    >
      <div
        className="m-tnm-sidebar__stop-card"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        style={
          Object {
            "cursor": "pointer",
          }
        }
        tabIndex={0}
      >
        <div
          className="m-tnm-sidebar__stop-card-header"
        >
          <a
            className="m-tnm-sidebar__stop-card-name"
          >
            Malden Center
          </a>
          <div
            className="m-tnm-sidebar__stop-distance"
          >
            288 ft
          </div>
        </div>
        <div
          className="m-tnm-sidebar__stop-card-description"
        >
          <span
            className="m-tnm-sidebar__stop-route-name"
          >
            <span
              className="m-tnm-sidebar__mode-icon"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "SVG",
                }
              }
            />
            <span>
              <a>
                Orange Line
              </a>
              
            </span>
          </span>
        </div>
        <div
          className="m-tnm-sidebar__stop-card-description"
        >
          <span
            className="m-tnm-sidebar__stop-route-name"
          >
            <span
              className="m-tnm-sidebar__mode-icon"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "SVG",
                }
              }
            />
            <a>
              Commuter Rail
            </a>
          </span>
        </div>
        <div
          className="m-tnm-sidebar__stop-card-description"
        >
          <span
            className="m-tnm-sidebar__stop-route-name"
          >
            <span
              className="m-tnm-sidebar__mode-icon"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "SVG",
                }
              }
            />
            <span>
              Bus: 
              <a>
                97
              </a>
              , 
              <a>
                99
              </a>
              , 
              <a>
                101
              </a>
              , 
              <a>
                104
              </a>
              , 
              <a>
                105
              </a>
              , 
              <a>
                106
              </a>
              , 
              <a>
                108
              </a>
              , 
              <a>
                131
              </a>
              , 
              <a>
                132
              </a>
              , 
              <a>
                136
              </a>
              , 
              <a>
                137
              </a>
              , 
              <a>
                411
              </a>
              , 
              <a>
                430
              </a>
              
            </span>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`StopsSidebar it returns null when there isn't data 1`] = `
<div
  className="m-tnm-sidebar"
  id="tnm-sidebar-by-stops"
>
  <div
    className="m-tnm-sidebar__fixed-header"
  >
    <div
      className="m-tnm-sidebar__fixed-header-inner"
    >
      <div
        className="m-tnm-sidebar__filter-bar"
      >
        <span
          className="m-tnm-sidebar__filter-header u-small-caps"
        >
          Filter
        </span>
        <div
          className="c-mode-filter__filter-btn-group"
        >
          <button
            aria-label="add filter by subway"
            className="btn btn-secondary btn-sm m-tnm-sidebar__filter-btn inactive"
            id="mode-button__subway"
            onClick={[Function]}
            type="button"
          >
            <span
              className="m-tnm-sidebar__mode-icon"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "SVG",
                }
              }
            />
            Subway
          </button>
          <button
            aria-label="add filter by bus"
            className="btn btn-secondary btn-sm m-tnm-sidebar__filter-btn inactive"
            id="mode-button__bus"
            onClick={[Function]}
            type="button"
          >
            <span
              className="m-tnm-sidebar__mode-icon"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "SVG",
                }
              }
            />
            Bus
          </button>
        </div>
        <div
          className="c-mode-filter__filter-btn-group"
        >
          <button
            aria-label="add filter by commuter_rail"
            className="btn btn-secondary btn-sm m-tnm-sidebar__filter-btn inactive"
            id="mode-button__commuter_rail"
            onClick={[Function]}
            type="button"
          >
            <span
              className="m-tnm-sidebar__mode-icon"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "SVG",
                }
              }
            />
            Rail
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="m-tnm-sidebar__inner"
  >
    <div
      className="m-tnm-sidebar__header"
    >
      <div
        className="m-tnm-sidebar__header-title"
      >
        <h2>
          Nearby Stops
        </h2>
        <div
          className="m-tnm-sidebar__view-change"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex={0}
        >
          Change view
        </div>
      </div>
    </div>
    <div
      className="m-tnm-sidebar__cards"
    >
      <div>
        No data
      </div>
    </div>
  </div>
</div>
`;
